test/test_linalg.py

```diff
@@ -3043,6 +3043,61 @@ def test_old_matrix_rank(self, device, dtype):
             self.assertEqual(torch.matrix_rank(aaT, True), np.linalg.matrix_rank(aaT.cpu().numpy(), True))
             self.assertEqual(torch.matrix_rank(aaT, 0.01, True), np.linalg.matrix_rank(aaT.cpu().numpy(), 0.01, True))
 
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    @precisionOverride({torch.float: 1e-03, torch.cfloat: 1e-03})
+    def test_multi_dot(self, device, dtype):
+        def check(*shapes):
+            tensors = [torch.randn(shape, device=device, dtype=dtype) for shape in shapes]
+            np_arrays = [tensor.cpu().numpy() for tensor in tensors]
+            res = torch.linalg.multi_dot(tensors).cpu()
+            ref = torch.from_numpy(np.array(np.linalg.multi_dot(np_arrays)))
+            self.assertEqual(res, ref)
+
+        # test for inputs with empty dimensions
+        check([0], [0])
+        check([2], [2, 0])
+        check([1, 0], [0])
+        check([0, 2], [2, 1])
+        check([2, 2], [2, 0])
+        check([2, 0], [0, 3])
+        check([0, 0], [0, 1])
+        check([4, 2], [2, 0], [0, 3], [3, 2])
+
+        # test variable output shapes
+        check([2], [2])
+        check([1, 2], [2])
+        check([2], [2, 1])
+        check([1, 2], [2, 1])
+        check([3, 2], [2, 4])
+
+        # test multiple input tensors
+        check([3], [3, 4], [4, 2], [2, 5], [5])
+        check([1, 2], [2, 2], [2, 3], [3, 1])
+
+        # test large tensors
+        check([10, 100], [100, 5], [5, 50])
+
+        # test discontiguous input
+        shapes = [[3, 2], [2, 2], [2, 3], [3, 4]]
```

Comment:
Let's add a `discontiguous` kwarg to make_tensor, which could be a separate PR but is pretty straightforward and will help write this PR (which will also provide a test case for the kwarg).

The discontiguous kwarg should return the same tensor the normal call would, except the last dimension should only have every other element populated. 

Possible implementation:

```
# given a tensor a
b = torch.repeat_interleave(a, 2, dim=-1)
return b[..., ::2]
```